src/concurrency/timestamp_ordering_transaction_manager.cpp

```diff
@@ -93,6 +93,10 @@ Transaction *TimestampOrderingTransactionManager::BeginTransaction() {
   auto eid = EpochManagerFactory::GetInstance().EnterEpoch(begin_cid);
   txn->SetEpochId(eid);
 
+  auto &log_manager = logging::LogManager::GetInstance();
+  log_manager.PrepareLogging();
+  log_manager.LogBeginTransaction(begin_cid);
```

Comment:
If we write this record here, what do we do if we abort the transaction? Will we still have to clean up the log somehow?